R/plotting/low_access_zones.R

```diff
@@ -0,0 +1,90 @@
+library("tidyverse")
+library("config")
+library("here")
+library("sf")
+
+# Read files ----
+
+file_path <- file.path(get("helmet_data"), get("present_scenario"), "agents.txt")
+agents <- read_delim(file_path, delim = "\t")
+
+file_path <- file.path(get("helmet_data"), get("baseline_scenario"), "agents.txt")
+agents0 <- read_delim(file_path, delim = "\t")
+
+file_path <- file.path(get("helmet_data"), get("projected_scenario"), "agents.txt")
+agents1 <- read_delim(file_path, delim = "\t")
+
+zones <- st_read(here("data", "helmet_zones_map.shp"))
+
+# Parameters for plotting ----
+
+limit_persons <- 30
+limit_txt <- paste0("Zones with over ", limit_persons, " persons")
+plot_areas <- c("helsinki_cbd", "helsinki_other", "espoo_vant_kau", "surrounding")
+
+# Calculate tour access ----
+
+agents <- agents %>%
+  filter(nr_tours > 0) %>%
+  mutate(tour_access = total_access / nr_tours)
+
+agents0 <- agents0 %>%
+  filter(nr_tours > 0) %>%
+  mutate(tour_access = total_access / nr_tours)
+
+agents1 <- agents1 %>%
+  filter(nr_tours > 0) %>%
+  mutate(tour_access = total_access / nr_tours)
+
+# Calc differences ----
+
+low_limit <- quantile(agents0$tour_access, probs = 0.05, na.rm = TRUE)[[1]]
+
+calc_low_access <- function(df, name){
+  df <- df %>%
+    mutate(low_access = (tour_access < low_limit)) %>%
+    group_by(number) %>%
+    summarise(!!name := sum(low_access, na.rm = TRUE),
```

Comment:
This is new notation for me! 😮 Can you teach me what `!!` and `:=` do here?

Reply:
Nevermind, I found a reference from `rlang` package (https://github.com/rstudio/cheatsheets/blob/master/tidyeval.pdf)! :)